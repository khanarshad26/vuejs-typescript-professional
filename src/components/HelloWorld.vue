<!-- eslint-disable prettier/prettier -->
<template>
  <div class="hello">
    <h1>{{ msg }}</h1>
    <input id="input" type="text" v-model="name" @input="handleInput"/>
    <button @click="handleInput">Button</button>
    <div v-if="showError">Error: Max character limit exceed of text input</div>
    <div v-if="!showError" > Name : {{name}}</div>
  </div>
</template>

<script lang="ts">
import { Vue } from "vue-class-component";
import { Prop } from "vue-property-decorator";
import { useStore } from "vuex";

export default class HelloWorld extends Vue {
  @Prop({ default: "This is a vue App" }) msg!: string;
  name = "";
  showError = false;
  store = useStore();
  // count = computed(() => this.store.state.count);
  // name = computed(() => this.store.state.name);
  increment = (): void => this.store.commit("increment");
  updateName = (newName: string): void =>
    this.store.commit("updateName", newName);

  handleInput(): void {
    if (this.name.length > 8) this.showError = true;
    else this.showError = false;
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
