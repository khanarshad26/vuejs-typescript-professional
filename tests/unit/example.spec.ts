import HelloWorld from "@/components/HelloWorld.vue";
import { shallowMount } from "@vue/test-utils";

describe("HelloWorld.vue", () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let wrapper: any;
  beforeEach(() => {
    wrapper = shallowMount(HelloWorld, {
      props: {
        msg: "default msg",
      },
    });
  });

  it("should render default snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders different prop.msg when passed", async () => {
    await wrapper.setProps({ msg: "test msg" });
    expect(wrapper.text()).toMatch("test msg");
  });

  it("should give error when max character limit exceed of text input", async () => {
    const input = await wrapper.find("#input");
    await input.setValue("arshadguyuygnu");
    expect(wrapper.text()).toContain(
      "Error: Max character limit exceed of text input"
    );
  });

  it("should render name if limit doesn't exceed", async () => {
    const input = await wrapper.find("#input");
    await input.setValue("arshad");
    expect(wrapper.text()).toContain("arshad");
  });

  it("should not render error when limit doesn't exceed", async () => {
    const input = await wrapper.find("#input");
    await input.setValue("arshad");
    expect(wrapper.vm.showError).toBeFalsy();
  });

  afterEach(() => {
    wrapper.unmount();
  });
});
