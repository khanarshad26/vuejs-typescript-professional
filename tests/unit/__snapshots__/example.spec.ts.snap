// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue should render default snapshot 1`] = `
<div
  class="hello"
>
  <h1>
    default msg
  </h1>
  <input
    id="input"
    type="text"
  />
  <button>
    Button
  </button>
  <!--v-if-->
  <div>
     Name : 
  </div>
</div>
`;
